### ISSUE TITLE
xfering to sliders and back has unconnected yarn

### ISSUE THREAD

**Issue by cat-Yu** [textiles-lab/knitout-live-visualizer#2]
<img width="733" alt="Screen Shot 2020-07-15 at 11 45 02 AM" src="https://user-images.githubusercontent.com/28764429/87566343-e2bf4e80-c690-11ea-8d51-95d87c821dce.png">
Image above produced by the code below:  

```javascript
const width = 4; // if width=2, sliders look correct

const knitout = require('knitout');
const carriers = ['1'];
const k = new knitout.Writer({carriers: carriers});
k.addHeader('Machine','SWGXYZ');
k.addHeader('Gauge','15');
k.inhook(carriers);

// tuck all needles
for (var i=0; i<width; i++){
    var b = i<(width/2) ? "f" : "b";
    k.tuck("+", b+i, carriers);
}

// one row of knits
for (var i=width/2-1; i>=0; i--){
    k.knit("-", "f"+i, carriers);
}
for (var i=width/2; i<width; i++){
    k.knit("+", "b"+i, carriers);
}

k.rack(1);

for (var i=width-1; i>=0; i--){
    var fb = i<(width/2) ? "f" : "b";
    var tb = i<(width/2) ? "bs" : "fs";
    if (i<(width/2)) k.xfer(fb+i, tb+(i-1));
    k.xfer(fb+i, tb+(i+1));
}

k.rack(0);

for (var i=width-1; i>=0; i--){
    var fb = i<(width/2) ? "bs" : "fs";
    var tb = i<(width/2) ? "f" : "b";
    if (i<(width/2)) k.xfer(fb+(i-1), tb+(i-1));
    k.xfer(fb+(i+1), tb+(i+1));
}

k.write('example.k');
```

**Comment by ixchow** [comment]
A more minimal test-case:
```
;!knitout-2
;;Carriers: 1
inhook 1
tuck + f1 1 
tuck + b2 1 
xfer f1 bs1
rack -2
xfer b2 f0
rack 0
xfer bs1 f1
```
![image](https://user-images.githubusercontent.com/1418494/88115849-23cdcc00-cb85-11ea-8d00-1260d895be6b.png)

**Comment by ixchow** [comment]
Fixed in 23b87ac1acd65f0ab2989aad449568f349ba355f.

![image](https://user-images.githubusercontent.com/1418494/88116256-31378600-cb86-11ea-9141-65735b628011.png)
